docs/reference/iam-auth-policy.md

```diff
@@ -0,0 +1,212 @@
+            <p>TargetRef points to the Kubernetes Gateway, HTTPRoute, or GRPCRoute resource that will have this policy attached.</p>
+            <p>This field is following the guidelines of Kubernetes Gateway API policy attachment.</p>
+         </td>
+      </tr>
+   </tbody>
+</table>
+<h3 id="application-networking.k8s.aws/v1alpha1.IAMAuthPolicyStatus">IAMAuthPolicyStatus</h3>
+<p>
+   (<em>Appears on:</em><a href="#application-networking.k8s.aws/v1alpha1.IAMAuthPolicy">IAMAuthPolicy</a>)
+</p>
+<div>
+   <p>IAMAuthPolicyStatus defines the observed state of IAMAuthPolicy.</p>
+</div>
+<table>
+   <thead>
+      <tr>
+         <th>Field</th>
+         <th>Description</th>
+      </tr>
+   </thead>
+   <tbody>
+      <tr>
+         <td>
+            <code>conditions</code><br/>
+            <em>
+            <a href="https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.24/#condition-v1-meta">
+            []Kubernetes meta/v1.Condition
+            </a>
+            </em>
+         </td>
+         <td>
+            <em>(Optional)</em>
+            <p>Conditions describe the current conditions of the IAMAuthPolicy.</p>
+            <p>Implementations should prefer to express Policy conditions
+               using the <code>PolicyConditionType</code> and <code>PolicyConditionReason</code>
+               constants so that operators and tools can converge on a common
+               vocabulary to describe IAMAuthPolicy state.
+            </p>
+            <p>Known condition types are:</p>
+            <ul>
+               <li>&ldquo;Accepted&rdquo;</li>
+               <li>&ldquo;Ready&rdquo;</li>
+            </ul>
+         </td>
+      </tr>
+   </tbody>
+</table>
+
+
+## IAMAauthPolicy Example
+
+```yaml
+apiVersion: application-networking.k8s.aws/v1alpha1
+kind: IAMAuthPolicy
+metadata:
+    name: test-iam-auth-policy
+spec:
+    targetRef:
+        group: "gateway.networking.k8s.io"
+        kind: HTTPRoute
+        name: my-route
+    policy: |
+        {
+            "Version": "2012-10-17",
+            "Statement": [
+                {
+                    "Effect": "Allow",
+                    "Principal": "*",
+                    "Action": "vpc-lattice-svcs:Invoke",
+                    "Resource": "*",
+                    "Condition": {
+                        "StringEquals": {
+                            "vpc-lattice-svcs:RequestHeader/header1": "value1"
+                        }
+                    }
+                }
+            ]
+        }
+```
+
+If you create the above IAMAuthPolicy in the k8s cluster, the `my-route` (and it's corresponding VPC Lattice service) will be attached with the given IAM auth policy. 
+Only HTTP traffic with header `header1:value1` will be allowed to access the `my-route`. Please check the [VPC Lattice documentation](https://docs.aws.amazon.com/vpc-lattice/latest/ug/auth-policies.html#auth-policies-common-elements) to get more detail on how lattice auth policy work.
```

Comment:
Didnt get what the point of this part.

Reply:
Do you mean remove the whole `## IAMAauthPolicy Example` section?